### ISSUE TITLE
[Phase 3] Deep Learning Model Architecture

### ISSUE THREAD

**Issue by Sakeeb91** [Sakeeb91/seismic-event-detection#4]
## Phase 3: Deep Learning Model Architecture

**Parent:** #1
**Depends on:** #2

### Objectives
Implement simplified EQTransformer architecture with multi-task output heads.

### Tasks

- [ ] Implement ResBlock for 1D convolutions
- [ ] Implement encoder (Conv + ResBlocks + Pool)
- [ ] Add Transformer encoder layers
- [ ] Implement three decoder heads (det, P, S)
- [ ] Verify output shapes match input (6000 samples)
- [ ] Test forward pass with batch input
- [ ] Count and verify parameters

### Files to Create/Modify

| File | Action | Description |
|------|--------|-------------|
| `src/models.py` | Create | SeismicDetector architecture |

### Code Reference

```python
class SeismicDetector(nn.Module):
    def __init__(self, in_channels: int = 3):
        super().__init__()

        # Encoder: Conv + ResBlocks + Pooling
        self.encoder = nn.Sequential(
            nn.Conv1d(in_channels, 8, 11, padding=5),
            nn.BatchNorm1d(8), nn.ReLU(),
            ResBlock(8), ResBlock(8), nn.MaxPool1d(2),
            # ... more layers
        )

        # Transformer
        encoder_layer = nn.TransformerEncoderLayer(
            d_model=64, nhead=8, dim_feedforward=128, batch_first=True
        )
        self.transformer = nn.TransformerEncoder(encoder_layer, num_layers=2)

        # Three decoder heads
        self.det_decoder = self._make_decoder(64, 1)
        self.p_decoder = self._make_decoder(64, 1)
        self.s_decoder = self._make_decoder(64, 1)

    def forward(self, x):
        enc = self.encoder(x)
        enc = self.transformer(enc.transpose(1, 2)).transpose(1, 2)
        return self.det_decoder(enc), self.p_decoder(enc), self.s_decoder(enc)
```

### Definition of Done

- [ ] ResBlock correctly adds residual connections
- [ ] Encoder downsamples then Transformer processes
- [ ] Three decoders upsample back to 6000 samples
- [ ] All outputs have sigmoid activation [0, 1]
- [ ] Forward pass: (batch, 3, 6000) -> 3x (batch, 1, 6000)

### Technical Notes

**For junior developers:**
- ResBlocks help gradient flow in deep networks
- Transformer captures long-range dependencies in waveform
- Three separate heads for multi-task learning
- Sigmoid outputs interpretable as probabilities
